lib/packages/docutils/rst.nim

```diff
@@ -1937,13 +1945,32 @@ proc parseEnumList(p: var RstParser): PRstNode =
     wildToken: array[0..5, int] = [4, 3, 3, 4, 3, 3]  # number of tokens
     wildIndex: array[0..5, int] = [1, 0, 0, 1, 0, 0]
       # position of enumeration sequence (number/letter) in enumerator
-  result = newRstNodeA(p, rnEnumList)
   let col = currentTok(p).col
   var w = 0
   while w < wildcards.len:
     if match(p, p.idx, wildcards[w]): break
     inc w
   assert w < wildcards.len
+  template checkAfterNewline =
+    let j = tokenAfterNewline(p, start=p.idx+1)
+    if p.tok[j].kind notin {tkIndent, tkEof} and
+        p.tok[j].col < p.tok[p.idx+wildToken[w]].col and
+        (p.tok[j].col > col or
+          (p.tok[j].col == col and not match(p, j, wildcards[w]))):
+      # check `result == nil` to avoid duplication of warning since for
+      # subsequent enum.items parseEnumList will be called second time
+      if result == nil:
+        let n = p.line + p.tok[j].line
+        rstMessage(p, mwRstStyle, ("\n  not enough indentation on line $2" &
```

Comment:
use `""" ... """` instead, more readable

Reply:
OK. Though I don't like to use `"""` without `dedent` — and it's not available.

Reply:
* what do you mean? it works if i add dedent
* also, it's cleaner to not add `"\n" &`, this works:
```
        let msg = """

          not enough indentation on line $2
              (if it's continuation of enumeration list),
          or no blank line after line $1 (if it should be the next paragraph),
          or no escaping \ at the beginning of line $1
              (if lines $1..$2 are a normal paragraph, not enum. list)""".dedent
```

Reply:
1. I mean bootstrap problem in #16699, which you root-caused as #16646.
2. yes, I prefer save 1 line of code

Reply:
just added `dedent` and tried to bootstrap, the error is still there:
```
rst.nim(1972, 3) template/generic instantiation of `checkAfterNewline` from here
rst.nim(1964, 26) Error: undeclared identifier: 'dedent'
FAILURE
```

Reply:
if you are OK with such workarounds let us try it: pushed to this PR.

Also because syntax `dedent*(s: string, count: Natural = indentation(s))` does not work for bootstrap, I had to split that `dedent` to `dedent(s, count)` and `dedent(s)`

Reply:
The workarounds around dedent are not worth it IMO.

Reply:
@Clyybber if you mean omitting separate proc `dedent(s)` then we need it anyway, it's the default option, it's usually being used without second argument. So we can't declare only `dedent(s, count)`.

Reply:
I mean calling dedent or unindent with an explicit count param, like so:
```nim

```
        let msg = """

          not enough indentation on line $2
              (if it's continuation of enumeration list),
          or no blank line after line $1 (if it should be the next paragraph),
          or no escaping \ at the beginning of line $1
              (if lines $1..$2 are a normal paragraph, not enum. list)""".dedent(10)

Reply:
this is fragile, people should be able to re-indent code without worrying about breaking semantics